ompi/mca/coll/base/coll_base_alltoall.c

```diff
@@ -63,56 +63,57 @@ mca_coll_base_alltoall_intra_basic_inplace(const void *rbuf, int rcount,
     ompi_datatype_get_true_extent ( rdtype, &true_lb, &true_ext);
     max_size = true_ext + ext * (rcount-1);
 
+    /* Initiate all send/recv to/from others. */
+    reqs = coll_base_comm_get_reqs(base_module->base_data, 2);
+    if( NULL == reqs ) { err = OMPI_ERR_OUT_OF_RESOURCE; line = __LINE__; goto error_hndl; }
```

Comment:
@bosilca 
Why not just returning OMPI_ERR_OUT_OF_RESOURCE here?
If you don't, you should check reqs!=NULL before calling free_reqs() at error_hndl
(or insert a check inside the routine free_reqs() as proposed in the first review)